### PR TITLE
added example with gulp

### PR DESCRIPTION
**Pull request by pazams**

### CODE REVIEW

File: examples/gulp/build/index.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<html>
```
Comment (RobLoach) [inline examples/gulp/build/index.html]: Should likely add the `build` directory to `.gitignore`.

Comment (Ajedi32) [inline examples/gulp/build/index.html]: That's not what all the other examples did. My guess is it's to make it possible for developers to see the result of the build here on GitHub without having to run the example for themselves.

Comment (RobLoach) [inline examples/gulp/build/index.html]: Ah, Gotcha. :+1:

File: examples/gulp/gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    connect = require('gulp-connect'),
+    metalsmith = require('metalsmith'),
+    markdown = require('metalsmith-markdown'),
+    permalinks = require('metalsmith-permalinks'),
+    templates = require('metalsmith-templates'),
+    watch = require('metalsmith-watch');
+
+    gulp.task('serve', function() {
+        connect.server();
+    });
```
Comment (Ajedi32) [inline examples/gulp/gulpfile.js]: Was the intent here to serve the entire project directory, including the source folder, gulpfile, etc? Wouldn't it make more sense to just serve the `build/` directory?

**Pull request closed without merging**